Fully specify room versions (#3432)
matrix-org/matrix-doc: content/rooms/v4.md

### Redactions

{{% rver-fragment name="v1-redactions" %}}

### Event format

The event format is the same as [room version 3](/rooms/v3#event-ids),

> however the event ID in the following example has been made to represent
> the changes in this room version.

Reviewer: ```suggestion
however the event IDs in the following example are updated to reflect
the changes in this room version.
```
